### PR TITLE
[Gluten-253] Fix merge operators issue when executing with DS V2 and there is only one partition

### LINKED ISSUE #253: Fix merge operators issue when executing with DS V2 and there is only one partition
**Describe the bug**
Fix merge operators issue when executing with DS V2 and there is only one partition.

**To Reproduce**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by zzcclp**
## What changes were proposed in this pull request?

Fix merge operators issue when executing with DS V2 and there is only one partition.

Close #253 .

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala [hunk 1]
```diff
@@ -205,6 +208,43 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
...
+
+    def transformChildren(plan: SparkPlan,
+                          basicScanExecTransformers: mutable.ListBuffer[BasicScanExecTransformer]
+                         ): Unit = {
+      if (plan != null && plan.isInstanceOf[TransformSupport]) {
+        if (plan.isInstanceOf[BasicScanExecTransformer]) {
+          basicScanExecTransformers.append(plan.asInstanceOf[BasicScanExecTransformer])
+        }
+        // according to the substrait plan order
+        if (plan.isInstanceOf[ShuffledHashJoinExecTransformer]) {
```
Comment (rui-mo) [inline jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala:225]: Is the special handling for other join also needed? Maybe change to HashJoinLikeExecTransformer and also consider SortMergeJoinExecTransformer.

File: jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala [hunk 2]
```diff
@@ -205,6 +208,43 @@ case class WholeStageTransformerExec(child: SparkPlan)(val transformStageId: Int
...
+            basicScanExecTransformers))
+        }
+      }
+    }
+
+    transformChildren(child, basicScanExecTransformers)
+    basicScanExecTransformers.toSeq.map( scan => {
+      scan match {
+        case op: BatchScanExecTransformer =>
+          op.partitions
```
Comment (rui-mo) [inline jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala:240]: Is `op.partitions` needed?

**Comment by rui-mo** [comment]
Thanks for your work!

File: jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala (hunk 2 above)
Comment (zzcclp) [reply jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala:240]: Removed.

File: jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala (hunk 1 above)
Comment (zzcclp) [reply jvm/src/main/scala/io/glutenproject/execution/WholeStageTransformerExec.scala:225]: Good suggestion, fixed.

**Review by rui-mo — APPROVED** [review]

**Comment by marin-ma** [comment]
LGTM.

**Review by marin-ma — APPROVED** [review]

**Pull request merged**